build-tools.php

```diff
@@ -0,0 +1,305 @@
+            ->setDescription('Creates the changelog between two versions.')
+
+            // the full command description shown when running the command with
+            // the "--help" option
+            ->setHelp('Create a changelog from two version branches or tags')
+
+            ->addArgument('project', InputArgument::REQUIRED, 'The development-collection prefix, either `flow` or `neos`')
+            ->addArgument('prevVersion', InputArgument::REQUIRED, 'The previous version, e.g. 6.2.14')
+            ->addArgument('version', InputArgument::REQUIRED, 'The version to generate the changelog for, e.g. 6.3.0')
+            ->addArgument('target', InputArgument::REQUIRED, 'The target file to write the changelog to')
+
+            ->addOption('buildUrl', null, InputOption::VALUE_OPTIONAL, 'The build URL to use in the commit message')
+            ->addOption('githubToken', null,InputOption::VALUE_OPTIONAL, 'To authenticate github calls and avoid API limits')
+            ->addOption('filter', null, InputOption::VALUE_OPTIONAL, 'A filter regex to apply to PR titles, e.g. "FEATURE" to only include features')
+            ->addOption('nocommit', null, InputOption::VALUE_NONE, 'Specify this if the changelog should not be committed')
+        ;
+    }
+
+    /**
+     * Do some cleanup on the PR message - strip some technical comments, convert links, cleanup newlines and fix escape sequences
+     */
+    protected function cleanupPrMessage(string $message): string
+    {
+        # Drop some footer lines from commit messages
+        $message = preg_replace('/^Change-Id: (I[a-f0-9]+)$/', '', $message);
+        $message = preg_replace('/^Releases?:.*$/', '', $message);
+        $message = preg_replace('/^Migration?:.*$/', '', $message);
+        $message = preg_replace('/^Reviewed-(by|on)?:.*$/', '', $message);
+        $message = preg_replace('/^Tested-by?:.*$/', '', $message);
+        $message = preg_replace('/<!--.*?-->\s*/s', '', $message);
+        $message = preg_replace('/\*\*Checklist\*\*.*?(- \[.\].*?[\r\n]+)+/s', '', $message);
+        $message = preg_replace('/\*\*(?:What I did|How I did it|How to verify it)\*\*[\n\s]+(?=(\*\*|$))/', '', $message);
+
+        # Link issues to Jira
+        $message = preg_replace('/(Fixes|Resolves|Related|Relates)?: (NEOS|FLOW)-([0-9]+)/', '* $1: `$2-$3 <https://jira.neos.io/browse/$2-$3>`_', $message);
+
+        # Link issues to GitHub
+        $message = preg_replace('/(Fixes|Solves|Resolves|Related(?:\sto)?|Relates|See):? #([0-9]+)/', "* $1: `#$2 <https://github.com/neos/{$this->project}-development-collection/issues/$2>`_", $message);
+        $message = preg_replace('/([a-zA-Z0-9]+\/[-.a-zA-Z0-9]+)#([0-9]+)/', '`#$2 <https://github.com/$1/issues/$2>`_', $message);
+        $message = preg_replace('/#([0-9]+)\s(?!<http)/', "`#$1 <https://github.com/neos/{$this->project}-development-collection/issues/$1>`_", $message);
```

Comment:
Link `neos/neos-development-collection#1234` as well as just `#1234` that are left-over